packages/web-app-files/src/mixins/actions/restore.ts

```diff
@@ -43,7 +51,126 @@ export default {
     ...mapMutations('runtime/spaces', ['UPDATE_SPACE_FIELD']),
     ...mapMutations(['SET_QUOTA']),
 
-    async $_restore_trigger({ resources }) {
+    getParentFolderFromResource(resource) {
+      return resource.path.slice(0, resource.path.lastIndexOf(resource.name))
```

Comment:
This is the same is `dirname`, isn't it? See https://nodejs.org/docs/latest/api/path.html#pathdirnamepath
You can import it via `import { dirname } from 'path'`

Reply:
I'd vote for removing the function and just using `dirname(resource.path)` in all the locations where you used this function before.